test/commands/deploy/pipeline.test.ts

```diff
@@ -136,4 +173,173 @@ describe('deploy pipeline', () => {
+        })
+        .command(['deploy:pipeline', '-p=testProject', '-b=testBranch'])
+        .it('correctly computes the source pipeline stage id when deploying to first stage in the pipeline', () => {
+          expect(fetchAndValidatePipelineStageStub.called).to.equal(true);
+          // verify we made the request
+          expect(requestMock.called).to.equal(true);
+          // now that we know the request was made
+          // we can get the call argument
+          // and validate its values
+          const requestArgument = requestMock.getCall(0).args[0] as HttpRequest;
+          expect(requestArgument.url).to.contain(REST_PROMOTE_BASE_URL);
+          const urlParams: string[] = requestArgument.url.split('/');
+          // since we are deploy to the first stage in the pipeline
+          // the source stage Id = 'Approved'
+          expect(urlParams[urlParams.length - 1]).to.equal('Approved');
+        });
+
+      test
+        .stdout()
+        .stderr()
+        .do(() => {
+          // mock the pipeline stage record
+          // this is not the first stage on the pipeline stage
+          // so it has a previous stage
+          pipelineStageMock = {
+            Id: secondStageId,
+            sf_devops__Branch__r: {
+              sf_devops__Name__c: 'mockBranchName',
+            },
+            sf_devops__Pipeline__r: {
+              sf_devops__Project__c: 'mockProjectId',
+            },
+            sf_devops__Pipeline_Stages__r: {
+              records: [
+                {
+                  Id: firstStageId,
+                  sf_devops__Branch__r: {
+                    sf_devops__Name__c: 'mockBranchName',
+                  },
+                  sf_devops__Pipeline__r: {
+                    sf_devops__Project__c: 'mockProjectId',
+                  },
+                },
+              ],
+            },
+          };
+          fetchAndValidatePipelineStageStub = sandbox
+            .stub(Utils, 'fetchAndValidatePipelineStage')
+            .resolves(pipelineStageMock);
+          requestMock = sinon.stub();
+        })
+        .command(['deploy:pipeline', '-p=testProject', '-b=testBranch'])
+        .it(
+          'correctly computes the source pipeline stage id when deploying to the second stage in the pipeline',
+          () => {
+            expect(fetchAndValidatePipelineStageStub.called).to.equal(true);
+            // verify we made the request
+            expect(requestMock.called).to.equal(true);
+            // now that we know the request was made
+            // we can get the call argument
+            // and validate its values
+            const requestArgument = requestMock.getCall(0).args[0] as HttpRequest;
+            expect(requestArgument.url).to.contain(REST_PROMOTE_BASE_URL);
+            const urlParams: string[] = requestArgument.url.split('/');
+            expect(urlParams[urlParams.length - 1]).to.equal(firstStageId);
+          }
+        );
+    });
```

Comment:
What about a test for checking that the test level default option is 'Default'?

Reply:
Good catch, I added it, thanks.

Reply:
Nice